test/test_archiveiterator.py

```diff
@@ -157,4 +174,58 @@ def test_corrects_wget_bug(self):
         with self._find_first_by_type('example-wget-bad-target-uri.warc.gz', 'response') as record:
             assert record.rec_headers.get('WARC-Target-URI') == 'http://example.com/'
 
+    def test_digests(self):
```

Comment:
do you think it would make sense to split this test into two different tests?
One for failing to validate a digest and one for validation of  a digest?

Reply:
Sure, I can do that. I see that I also have a bunch of duplicated 3-line sections that I can stick in a function.

Reply:
(I also see that I missed adding some tests to test_limitreader for the new digest helper functions in limitreader.py)